### ISSUE TITLE
Request to add CSS #ID's to success cases

### ISSUE THREAD

**Issue by ahmadabugosh** [Giveth/giveth-dapps-v2#1731]
Hey @MoeNick and team, 

**Problem**
As I have started to set up Analytics tracking for our site, I noticed that it's a bit hard to track conversion events from our platform due to how our website CSS is structured. Since we don't use CSS ID's in any of our "success cases" (such as a successful donation, or project creation), I had to manually take the CSS class selector instead. 

See below what I mean: 
<img width="1409" alt="Screen Shot 2022-10-23 at 12 49 24 AM" src="https://user-images.githubusercontent.com/4855928/199105525-15813baa-1ffd-430d-a30b-3f40ee69ec3a.png">

If I want to detect if a Donation is made as an Analytics event, I have to create an event that says "if this element with the following CSS class is visible, then a Donation took place".

This works, however, if we ever change the CSS selector for any reason it would stop counting those events. 

For example, now when a donation takes place, I check if "Element Visible = true" for the CSS selector"h4.sc-hAZoDl.sc-de5ab0a6-2.bwTdqR.kQxyye"

**Solution** 

What I recommend is to start adding CSS ID's to each of the success cases. For example, on the place where it says "You're a giver now!" make the CSS be something like

`<h4 id="donation-success">You're a giver now!</h4>` 

We should then add ID's for all of our "success messages" on the platform, including:
- Donation success
- Project creation
- Project verification submitted
- GIVfarm staking
- GIVpower staking
- GIVpower boosting (when it's active)
-  Any other or new success cases. 

In short, I think it should be added to the dev pipeline to always add an ID to success messages, that way it would be very easy to track those actions in Analytics and define goals and conversion rates.

**Comment by divine-comedian** [comment]
@ahmadabugosh  is this still relevant? You're referring for using google analytics corroect/

**Comment by ahmadabugosh** [comment]
@divine-comedian Yes, I think this still would be useful, especially for successful donations to track this in Google Analytics properly!

**Comment by MohammadPCh** [comment]
@kkatusic can you take it up? thanks.

**Comment by kkatusic** [comment]
> @kkatusic can you take it up? thanks.

I will thx ;)

**Comment by divine-comedian** [comment]
@ahmadabugosh  - this should be done, do you need anything else to make it useful?

**Comment by kkatusic** [comment]
> @ahmadabugosh - this should be done, do you need anything else to make it useful?

@ahmadabugosh need only information when this is put online to setup on analytics.

**Comment by divine-comedian** [comment]
I was checking around in GA for these new events and I don't see them in our existing event list - did we ever QA that adding these CSS ID actually work? 

I think we'll need to test in GTM and GA before we can call this one done

**Comment by maryjaf** [comment]
> * - GIVpower staked => id=“givpowe-staked”

After success stake I could only  see this ID: 
![image](https://github.com/Giveth/giveth-dapps-v2/assets/111529185/6c17bc1a-0819-4c54-bf71-0e1cca75d7ea)

after which actions  I should see these Ids? after staking from givfarm I only see "givpower-staked" id 
 - GIVfarm staking => id=“givfarm-staking”
 - GIVpower staking => id=“givpowe-staking”
 - GIVfarm staked => id=“givfarm-staked”
 - RegenFarm staking => id=“regenfarm-staking”
 - RegenFarm staked => id=“regenfarm-staked”

**Comment by divine-comedian** [comment]
@kkatusic could you clarify where you ended up inserting these id's - it isn't clear the difference between `staking` and `staked` and the difference between GIVfarm and GIVpower

**Comment by kkatusic** [comment]
@maryjaf, cc @divine-comedian , for staking and staked you cna check this title in modal window. Staking is step before staked:

![middle-step_2C3FD88F](https://github.com/Giveth/giveth-dapps-v2/assets/1242145/c80886c5-383c-4d74-b746-6207acb96885)

also it was time of the regen I think and I add option for that also:

```
`// preffix property for heading elements used by analytics

const idPropertyPreffix = regenStreamType
? 'regenfarm'
: isGIVpower
	? 'givpower'
	: '';`
```

I don't know how to enable _regen_ on staging server to test, @RamRamez can you help

For GIVPower is this condition:

```
const isGIVpower =
		type === StakingType.GIV_GARDEN_LM ||
		type === StakingType.GIV_UNIPOOL_LM;
```

I predicted the ids, haven't tested GIVPower for GIV_GARDEN_LM or GIV_UNIPOOL_LM that

**Comment by maryjaf** [comment]
> @maryjaf, cc @divine-comedian , for staking and staked you cna check this title in modal window. Staking is step before staked:

Thanks for your explanation @kkatusic , I've checked GIVpower-staking  in stake flow and it is passed like as GIVpower-staked
and what about GIVfarm -staked and staking?

**Comment by kkatusic** [comment]
@maryjaf have you this enabled in your account I can't access to it?

![Screenshot 2024-07-11 at 11 33 07](https://github.com/Giveth/giveth-dapps-v2/assets/1242145/d5489bcc-131b-43fd-80a7-0afc66cd029e)

**Comment by maryjaf** [comment]
> @maryjaf have you this enabled in your account I can't access to it?
> 
> ![Screenshot 2024-07-11 at 11 33 07](https://private-user-images.githubusercontent.com/1242145/347763837-d5489bcc-131b-43fd-80a7-0afc66cd029e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2OTA3NzMsIm5iZiI6MTcyMDY5MDQ3MywicGF0aCI6Ii8xMjQyMTQ1LzM0Nzc2MzgzNy1kNTQ4OWJjYy0xMzFiLTQzZmQtODBhNy0wYWZjNjZjZDAyOWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTFUMDkzNDMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODZlMTBlY2ZiMGViZWIzMjJmNjhlMGQzZmRiYzY5YTI3M2EzOTMyYTFmYWNkYTc2YjhhODJiYjMyNDgzZDdjYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.BHJzVasjKeHdJTU4F5PGoncs6NBxeVKIZZkRMuny6RQ)

It is disabled for me too

**Comment by divine-comedian** [comment]
We don't use GIVfarm or regenFarms anymore - let's not break our heads on this issue and we can leave it out of scope for QA requirements.

**Comment by kkatusic** [comment]
> We don't use GIVfarm or regenFarms anymore - let's not break our heads on this issue and we can leave it out of scope for QA requirements.

Ids are there if we used regen or farm in future

**Comment by kkatusic** [comment]
@divine-comedian we are done with this?
